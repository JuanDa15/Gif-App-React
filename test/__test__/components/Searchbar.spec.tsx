import SearchBar from "../../../src/components/Searchbar"
import { fireEvent, render, screen } from "@testing-library/react"

describe('Test for <Searchbar />', () => {
  const handler = (value: string) => {value}
  test('should match with the snapshot', () => {
    const { container } = render(<SearchBar onSearch={handler} />)
    expect(container).toMatchSnapshot()
  })
  test('should change the input value', () => {
    const newInputValue = 'dragon ball';
    render(<SearchBar onSearch={handler} />)
    const input: HTMLInputElement = screen.getByRole('search') as HTMLInputElement;
    fireEvent.input(input, { target: { value: newInputValue}})
    expect(input.value).toBe(newInputValue);
  })
  test('should call the submit Fn', () => {
    const newInputValue = 'dragon ball';
    const mockedHandler = jest.fn(handler)
    render(<SearchBar onSearch={mockedHandler} />)
    
    const input = screen.getByRole('search') as HTMLInputElement;
    const form = screen.getByRole('searchbox') as HTMLFormElement;
    
    fireEvent.input(input, { target: { value: newInputValue}})
    fireEvent.submit(form)

    expect(mockedHandler).toHaveBeenCalledWith(newInputValue)
    expect(input.value).toBe('')
  })
  test('should not call the handler Fn', () => {
    const newInputValue = '';
    const mockedHandler = jest.fn(handler)
    render(<SearchBar onSearch={mockedHandler} />)

    const input = screen.getByRole('search') as HTMLInputElement;
    const form = screen.getByRole('searchbox') as HTMLFormElement;

    fireEvent.input(input, { target: { value: newInputValue}})
    fireEvent.submit(form)

    expect(mockedHandler).not.toHaveBeenCalled()
  })
})