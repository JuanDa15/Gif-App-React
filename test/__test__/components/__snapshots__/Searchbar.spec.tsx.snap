// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for <Searchbar /> should match with the snapshot 1`] = `
<div>
  <form
    class="flex justify-center items-center h-12"
    role="searchbox"
  >
    <input
      class="w-full h-full px-4 py-2 bg-gray-900 rounded-md focus:outline-purple-700 focus:outline-2 focus:outline rounded-r-none"
      placeholder="Search..."
      role="search"
      type="search"
      value=""
    />
    <button
      class="w-12 h-12 rounded-md bg-gray-900 hover:bg-purple-700 text-white grid place-items-center rounded-l-none transition-colors disabled:opacity-80"
      data-testid="search-button"
      disabled=""
      role="button"
      type="submit"
    >
      <svg
        class="w-6 h-6"
        fill="none"
        stroke="currentColor"
        stroke-width="1.5"
        viewBox="0 0 24 24"
      >
        <path
          d="M21 21l-5.197-5.197m0 0A7.5 7.5 0 105.196 5.196a7.5 7.5 0 0010.607 10.607z"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
    </button>
  </form>
</div>
`;
